src/sdi-notify.c

```diff
@@ -53,16 +168,11 @@ static void show_pending_update_notification(SdiNotify *self,
   }
   g_notification_set_default_action_and_target(
       notification, "app.close-notification", "s", "pending-update");
-  g_notification_add_button_with_target(
-      notification, "Close", "app.close-notification", "s", "pending-update");
-  g_autoptr(GVariantBuilder) builder =
-      g_variant_builder_new(G_VARIANT_TYPE("as"));
-  for (; snaps != NULL; snaps = snaps->next) {
-    SnapdSnap *snap = (SnapdSnap *)snaps->data;
-    g_variant_builder_add(builder, "s", snapd_snap_get_name(snap));
-  }
-  GVariant *values = g_variant_builder_end(builder);
-  g_notification_add_button_with_target_value(notification, "Ignore",
+  g_notification_add_button_with_target(notification, _("Close"),
+                                        "app.close-notification", "s",
+                                        "pending-update");
+  GVariant *values = get_snap_list(snaps);
```

Comment:
Could use a single line again, to avoid the confusion about the floating reference.

Reply:
Again, good point. Done too.